src/Host/Client/Impl/rtvs/R/util.R

```diff
@@ -194,3 +194,18 @@ rmarkdown_publish <- function(blob_id, output_format, encoding) {
     rmarkdown::render(rmdpath, output_format = output_format, output_file = output_filepath,  output_dir = tempdir(), encoding = encoding);
     readBin(output_filepath, 'raw', file.info(output_filepath)$size);
 }
+
+package_lock_state <- function(package_name, lib_path) {
```

Comment:
If you're making this remote, it will be better to add another 2 methods here - `package_install` and `package_uninstall` and call this function from them directly. Both functions can return if they have succeeded or failed because of the lock.